File: 백준/Silver/28278. 스택 2/스택 2.py
from collections import deque
import sys

stack = deque()
N = int(sys.stdin.readline().strip())

for _ in range(N):
    x = list(map(int, sys.stdin.readline().split()))
    if x[0] == 1:
        stack.append(x[1])

    elif x[0] == 2:
        if stack:
            print(stack.pop())
        else:
            print(-1)

    elif x[0] == 3:
        print(len(stack))

    elif x[0] == 4:
        if stack:
            print(0)
        else:
            print(1)

    elif x[0] == 5:
        if stack:
            print(stack[-1])
        else:
            print(-1)

File: 백준/Silver/28278. 스택 2/스택 2.java
import java.io.*;
import java.util.*;

public class Main {

    public static void main(String[] args) throws IOException {

        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringBuilder sb = new StringBuilder();
        Stack<Integer> stack = new Stack<>();

        int N = Integer.parseInt(br.readLine());

        for (int i = 0; i < N; i++) {
            StringTokenizer st = new StringTokenizer(br.readLine(), " ");
            switch (Integer.parseInt(st.nextToken())) {
                case 1:
                    stack.push(Integer.parseInt(st.nextToken()));
                    continue;
                case 2:
                    if (!stack.isEmpty()) {
                        sb.append(stack.pop()).append("\n");
                        continue;
                    } else {
                        sb.append(-1).append("\n");
                        continue;
                    }
                case 3:
                    sb.append(stack.size()).append("\n");
                    continue;
                case 4:
                    if (stack.isEmpty()) {
                        sb.append(1).append("\n");
                        continue;
                    } else {
                        sb.append(0).append("\n");
                        continue;
                    }
                case 5:
                    if (!stack.isEmpty()) {
                        sb.append(stack.peek()).append("\n");
                        continue;
                    } else {
                        sb.append(-1).append("\n");
                        continue;
                    }
            }
        }
        System.out.println(sb);
    }
}